.github/workflows/ci.yml

```diff
@@ -10,30 +10,40 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+
       - name: Setup go-task
         uses: arduino/setup-task@v1
         with:
           version: 3.x
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.22.x'
+          go-version: '1.24.x'
           token: ${{ secrets.GITHUB_TOKEN }}
           cache-dependency-path: |
             datatrails-common-api/go.sum
+
       - name: Setup protobuf
         uses: arduino/setup-protoc@v2
         with:
           version: "24.3"
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup protoc-gen-doc
```

Comment:
do we need to roll this actions change out to forestrie and go-datatrails-merklelog etc ? Or can those happen when well feel like it ?

Reply:
all is versioned so thos repos will not see any changes in this repo until we release this and update the dependency. 
When updating the dependency in (say) go-common-avid-api for example the tools must be installed using the 'go get -tool' command which will setup the go.mod file to manage all tools locally using the new 'tool' section. The one exception is that we must install protoc-gen-doc using the go install command with a version number which does not touch go.mod.

So yes we must roll similar changes out to other repos but we can do it when we feel like it